aws/table_aws_s3_bucket.go

```diff
@@ -318,25 +320,64 @@ func listS3Buckets(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateDa
 	return nil, nil
 }
 
-func getS3BucketEventNotificationConfigurations(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
-	// Bucket location will be nil if getBucketLocation returned an error but
-	// was ignored through ignore_error_codes config arg
-	if h.HydrateResults["getBucketLocation"] == nil {
-		return nil, nil
+func doGetBucketRegion(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, bucket string) (string, error) {
+	// Have we already resolved and cached the bucket name?
+	c, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_s3_bucket.doGetBucketRegion", "get_common_columns_error", err)
+		return "", err
+	}
+	commonColumnData := c.(*awsCommonColumnData)
+	cacheKey := "getBucketRegion/" + commonColumnData.Partition + "/" + bucket
+	if cachedData, ok := d.ConnectionManager.Cache.Get(cacheKey); ok {
+		return cachedData.(string), nil
+	}
+
+	// The most reliable way to discover the region of an S3 bucket is to make an unauthenticated HTTP HEAD request.
+	// See https://github.com/aws/aws-sdk-go/issues/356#issuecomment-132707340
+	// Using the S3 path style allows to query buckets with dots in their name.
+	// Not doing so with such buckets causes `tls: failed to verify certificate: x509: certificate is valid for *.s3.amazonaws.com, s3.amazonaws.com, not www.somedomain.com.s3.amazonaws.com (SQLSTATE HV000)` errors.
+	// FIXME: do we also want to implement non S3 path style? It may help avoiding rate limiting, but given the above limitation,
+	// it may be better to define a default Steampipe limiter once actual AWS limits are discovered.
+	resp, err := http.Head(fmt.Sprintf("https://s3.amazonaws.com/%s", bucket))
```

Comment:
@pdecat I think we need to handle other partitions, like US GovCloud, China, ISO-B, etc., else this request will return a 404 code I believe. The URL could be adjusted based on the `commonColumnData.Partition`

Reply:
That's a very good point, and apparently only buckets in standard AWS partition are exposed using this global endpoint. I do not have access to buckets in GovCloud and ISO partitions to verify this though. Can make some tests against buckets in China, but that won't cover all cases anyway.

Probably a definitive argument to switch to using the `manager.GetBucketRegion()` as [suggested by @vpartington](https://github.com/turbot/steampipe-plugin-aws/pull/2082#issuecomment-2051419905) a few weeks ago.

FWIW, it seems the [terraform provider does not support the ISO partitions](https://github.com/hashicorp/terraform-provider-aws/issues/18593) either.

Reply:
@pdecat We can do some testing against GovCloud buckets within our team, but it seems like switching to `manager.GetBucketRegion()` would be a good change if that function natively supports other partitions (even if it's not all of them, like ISO).

Would you be able to update this PR to use that function instead? Sorry, I don't want to drag this PR out more (mostly due to my slow response times), but we do have some users in non-commercial partitions at the moment.

If you need any help implementing/testing that change, please let us know, thanks!

Reply:
Done! Testing in progress...